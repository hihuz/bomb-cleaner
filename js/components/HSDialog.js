import React from 'react';

const HSDialog = ({ closeHSDialog, opened }) => (
  <div className={`dialog${opened?' opened':''}`}>
    <button className="dialog-close" onClick={closeHSDialog}>x</button>
    lu
  </div>
);

export default HSDialog;
