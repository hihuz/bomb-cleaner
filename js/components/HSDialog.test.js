import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import HSDialog from './HSDialog';

const testHS = {
  easy: [
    { name: 'Bob', time: 45, date: 'foo' },
    { name: 'Bah', time: 30, date: 'bar' },
  ],
  medium: [
    { name: 'Boo', time: 20, date: 'baz' },
  ],
  hard: []
};

test('Snapshot', () => {
  const component = shallow(<HSDialog highScores={testHS} />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Should call the closeDialog prop when the dialog-close button is clicked', () => {
  const clickFn = jest.fn();
  const component = shallow(<HSDialog closeDialog={clickFn} highScores={testHS} />);
  component.find('button').simulate('click');
  expect(clickFn).toHaveBeenCalledTimes(1);
});

test('Should display the correct number of highscores based off highScores prop', () => {
  const component = shallow(<HSDialog highScores={testHS} />);
  const lists = component.find('ol');
  const easyList = lists.at(0);
  const mediumList = lists.at(1);
  const hardList = lists.at(2);
  expect(easyList.find('li').length).toEqual(testHS.easy.length);
  expect(mediumList.find('li').length).toEqual(testHS.medium.length);
  expect(hardList.find('li').length).toEqual(testHS.hard.length);
});

test('Should display the props of each provided highscore correctly', () => {
  const component = shallow(<HSDialog highScores={testHS} />);
  const hs = component.find('ol').at(0).find('li').at(0);
  expect(hs.find('.hs-name').text()).toEqual(testHS.easy[0].name);
  expect(hs.find('.hs-time').text()).toEqual(`${testHS.easy[0].time} secs`);
  expect(hs.find('.hs-date').text()).toEqual(testHS.easy[0].date);
});
