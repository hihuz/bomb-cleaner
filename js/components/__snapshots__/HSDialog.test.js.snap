exports[`test Snapshot 1`] = `
<div>
  <button
    className="dialog-close">
    <i
      className="icon-close" />
  </button>
  <div
    className="hs-row">
    <h2>
      Easy
    </h2>
    <ol>
      <li
        className="hs">
        <div
          className="hs-name">
          Bob
        </div>
        <div
          className="hs-time">
          45
           secs
        </div>
        <div
          className="hs-date">
          foo
        </div>
      </li>
      <li
        className="hs">
        <div
          className="hs-name">
          Bah
        </div>
        <div
          className="hs-time">
          30
           secs
        </div>
        <div
          className="hs-date">
          bar
        </div>
      </li>
    </ol>
  </div>
  <div
    className="hs-row">
    <h2>
      Medium
    </h2>
    <ol>
      <li
        className="hs">
        <div
          className="hs-name">
          Boo
        </div>
        <div
          className="hs-time">
          20
           secs
        </div>
        <div
          className="hs-date">
          baz
        </div>
      </li>
    </ol>
  </div>
  <div
    className="hs-row">
    <h2>
      Hard
    </h2>
    <ol />
  </div>
</div>
`;
